### ISSUE TITLE
two stage denosing

### ISSUE THREAD

**Issue by caichuang0415** [lllyasviel/IC-Light#15]
Thanks for your wonderful work!
I have a question that, in your codes, the process function contains two stage denosing steps, I don't know why it needs two.
![image](https://github.com/lllyasviel/IC-Light/assets/50684798/c7397385-1d51-4122-8b81-28ea30e69561)

**Comment by Lalimec** [comment]
hires fix?

**Comment by caichuang0415** [comment]
> hires fix?

maybe

**Comment by V1sionVerse** [comment]
Yes, that second step is just the upscaler. I have modified my local copy so I can bypass it entirely because it runs much faster without the upscaler and doesn't mess with the original image nearly as much. It also eats up a ton of VRAM that you can use for larger generation dimensions instead.

**Comment by bragilee** [comment]
> Yes, that second step is just the upscaler. I have modified my local copy so I can bypass it entirely because it runs much faster without the upscaler and doesn't mess with the original image nearly as much. It also eats up a ton of VRAM that you can use for larger generation dimensions instead
> 
> You simply add the line `return pixels` after the line `pixels = pytorch2numpy(pixels)` and that bypasses the upscaler.

I only saw t2i() and i2i() pipelines. Is that a older version? 

t2i_pipe = StableDiffusionPipeline(
    vae=vae,
    text_encoder=text_encoder,
    tokenizer=tokenizer,
    unet=unet,
    scheduler=dpmpp_2m_sde_karras_scheduler,
    safety_checker=None,
    requires_safety_checker=False,
    feature_extractor=None,
    image_encoder=None
)
